lib/glific/providers/gupshup_enterprise/template.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Glific.Providers.GupshupEnterprise.Template do
+  @moduledoc """
+  Module for handling template operations specific to Gupshup
+  """
+
+  alias Glific.{
+    Partners,
+    Repo,
+    Settings.Language,
+    Templates
+  }
+
+  @doc """
+  Import pre approved templates when BSP is GupshupEnterprise
+  """
+  @spec import_enterprise_templates(non_neg_integer(), String.t()) :: {:ok, any}
+  def import_enterprise_templates(organization_id, data) do
+    {:ok, stream} = StringIO.open(data)
+    organization = Partners.organization(organization_id)
+
+    stream
+    |> IO.binstream(:line)
+    |> CSV.decode(headers: true, strip_fields: true)
+    |> Enum.map(fn {_, data} -> import_approved_templates(data) end)
+    |> Templates.do_update_hsms(organization, :gupshup_enterprise)
+
+    {:ok, %{message: "All templates have been added"}}
+  end
+
+  @spec import_approved_templates(map()) :: map()
+  defp import_approved_templates(template),
+    do: %{
+      "id" => Ecto.UUID.generate(),
+      "data" => template["Body"],
+      "meta" => get_example_body(template["Body"]),
+      "category" => "ALERT_UPDATE",
+      "elementName" => template["Template Name"],
+      "languageCode" => get_language(template["Language"]),
+      "templateType" => template["Type"],
+      "status" => get_status(template["Status"]),
+      "enterprise_id" => template["Template Id"]
+    }
+
+  @spec get_language(String.t()) :: String.t()
+  defp get_language(label_locale) do
+    {:ok, language} = Repo.fetch_by(Language, %{label_locale: label_locale})
+    language.locale
+  end
+
+  @spec get_status(String.t()) :: String.t()
```

Comment:
Let's have a status map with key-value pair 
status_map
Map.get(status, "PENDING")

Reply:
it does not have default value then until nil check is added. Using function handle default value `PENDING` in a clean way

Reply:
changed